### ISSUE TITLE
constness violation with modify-assign operators

### ISSUE THREAD

**Issue by LukasBanana** [Microsoft/DirectXShaderCompiler#53]
I used this online interface, which uses the latest DirectX Shader Compiler as far as I know, to verify my assumption: [tryhlsl.azurewebsites.net/](http://tryhlsl.azurewebsites.net/)

The following shader code should output a compiler error, but instead the modify-assign operation is ignored by the compiler:
```
float4 PSMain(PSInput input) : SV_TARGET
{
    const float c = 2.0;
    c += 3.0; // ERROR (ignored by the compiler)
    return (float4)c;
}
```
While the code above is compiled, and the output value is a constant 2.0 (which is obviously wrong),
the following code is rejected by the compiler with an error output:
```
float4 PSMain(PSInput input) : SV_TARGET
{
    const float c = 2.0;
    c = c + 3.0; // ERROR (correctly rejected by the compiler)
    return (float4)c;
}
```
This bug is present in the _fxc_ and also in the new _dxc_

**Comment by python3kgae** [comment]
Fixed with https://github.com/Microsoft/DirectXShaderCompiler/commit/26550b7bf4ad985f0c3cdef4d22ee79b2884c230

Thanks
Xiang

**Comment by dmpots** [comment]
Fixed with 26550b7
